Drop obsolete note that compiling with Qt 6 doesn't work

michaelweghorn opened the pull request:
Building with Qt 6 using the `-DUSE_QT6=ON` CMake flag works fine by now, so drop the note saying it doesn't.

B-Interactive commented:
I've yet to successfully get this working under Arch based Linux, referencing QT6 based dependencies.  I had assumed it was still a work in progress.  If it's working, I'd be interested to know more about the build requirements for that, because presently there are issues under Wayland with the QT5 build.

michaelweghorn commented:
> I've yet to successfully get this working under Arch based Linux, referencing QT6 based dependencies. I had assumed it was still a work in progress. If it's working, I'd be interested to know more about the build requirements for that, because presently there are issues under Wayland with the QT5 build.

I'm building on Debian testing, with system Qt 6 (currently: Qt 6.6.2), using this CMake command to configure:

    cmake -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=/home/michi/temp/bibletime -DBUILD_HANDBOOK_HTML=ON -DBUILD_HANDBOOK_HTML_LANGUAGES=en -DBUILD_HANDBOOK_PDF=OFF -DBUILD_HOWTO_HTML=OFF -DBUILD_HOWTO_PDF=OFF -DUSE_QT6=ON .

It runs fine on Wayland for me - both, the Qt 5 and the Qt 6 based build, i.e. with or without passing `-DUSE_QT6` to CMake.

Do you want to give more details on the issues you're encountering?

B-Interactive commented:
I appreciate the details you provided there.  Just knowing it should work, actually gave me drive to persist :grin: 

It is now successfully building using Qt6.  The issue I encountered was the paths for `lrelease` referenced in [cmake/BTUITranslation.cmake ](https://github.com/bibletime/bibletime/blob/b3ff1d542754a87aae44a4bf1f2bb4739b13f191/cmake/BTUITranslation.cmake#L4) don't account for the Qt6 version, at least not on an Arch based distro.  It wasn't finding it and failing to build.

On Arch, `lrelease` and `lupdate` from [`qt6-tools`](https://archlinux.org/packages/extra/x86_64/qt6-tools/) are located:
```
/usr/lib/qt6/bin/lupdate
/usr/lib/qt6/bin/lrelease
```
However, if the [`qt5-tools`](https://archlinux.org/packages/extra/x86_64/qt5-tools/) are installed on the system, cmake will find that version under `/usr/bin/lrelease`, and produce a successful build, albeit using some Qt5 tools to do it.

The simple fix was to add these two extra cmake commands, manually pointing to the correct location:
```
-DQT_LUPDATE_EXECUTABLE=/usr/lib/qt6/bin/lupdate \
-DQT_LRELEASE_EXECUTABLE=/usr/lib/qt6/bin/lrelease \
```

I'd suggest an update of [cmake/BTUITranslation.cmake ](https://github.com/bibletime/bibletime/blob/b3ff1d542754a87aae44a4bf1f2bb4739b13f191/cmake/BTUITranslation.cmake#L4) to account for Qt6 locations.

B-Interactive commented:
With respect to the Wayland+Qt5 issue I mentioned, I apologise, I'll see if I can verify that with further testing.  If it's something I can replicate and pin down, I'll open an issue properly.

michaelweghorn commented:
> I'd suggest an update of [cmake/BTUITranslation.cmake ](https://github.com/bibletime/bibletime/blob/b3ff1d542754a87aae44a4bf1f2bb4739b13f191/cmake/BTUITranslation.cmake#L4) to account for Qt6 locations.

Thanks for providing more details. I could reproduce that Qt 5 tools were still used, which I hadn't noticed before. https://github.com/bibletime/bibletime/pull/478 is my suggestion to make use of the Qt 6 versions when building with `-DUSE_QT6` (and for addressing some issues seen while working on that).

jaakristioja reviewed (approved):
Good catch! Thanks! :-)
